docs/content/5.recipes/4.custom-session-data.md

```diff
@@ -3,55 +3,115 @@
+    CredentialsProvider.default({
+    ...
+      async authorize(credentials: any) {
+        const data = {
+          user: {
+            name: 'John Doe',
+            email: 'john@email.com',
+            id: 1,
+            role: 'admin'
+          },
+          token: '123456890',
+        }
+
+        if (data.user) {
+          const u = {
+            id: data.user.id,
+            name: data.user.username,
+            email: data.user.email,
+            access_token: data.token, // additional field
+            role: response.user.role // additional field
+          };
+          return u;
+        } else {
+          throw createError({
+            statusCode: 403,
+            statusMessage: "Credentials not working",
+          });
+        }
+      },
+    )}
+   ]
+   ... // other config
+
+```
+We can see the new fields in the return object `u` `role` and `access_token`. Now we need to pass them in the callbacks.
+
+## Add callbacks to the `NuxtAuthProvider` to alter the session data
 
 First, you must add jwt and session callbacks to [the `NuxtAuthHandler`](/nuxt-auth/configuration/nuxt-auth-handler):
 ```ts
 // file: ~/server/api/auth/[...].ts
 export default NuxtAuthHandler({
+  // ... remainder of your config ...
   callbacks: {
     // Callback when the JWT is created / updated, see https://next-auth.js.org/configuration/callbacks#jwt-callback
-    async jwt({ token, account, profile }) {
-      // Persist the OAuth access_token and or the user id to the token right after signin
-      if (account) {
-        token.accessToken = account.access_token
-        token.id = profile.id
+    jwt: async ({token, user}) => {
+      const isSignIn = user ? true : false;
+      if (isSignIn) {
+        token.jwt = user ? (user as any).access_token || '' : '';
+        token.id = user ? user.id || '' : '';
+        token.role = user ? (user as any).role || '' : '';
       }
-      return token
+      return Promise.resolve(token);
     },
-
     // Callback whenever session is checked, see https://next-auth.js.org/configuration/callbacks#session-callback
-    async session({ session, token, user }) {
-      // Send properties to the client, like an access_token and user id from a provider.
-      session.accessToken = token.accessToken
-      session.user.id = token.id
-
-      return session
-    }
+    session: async ({session, token}) => {
+      (session as any).role = token.role;
+      (session as any).uid = token.id;
+      return Promise.resolve(session);
+    },
   },
   ..., // other config
   providers: [...], // Your provider config
 })
 ```
+Notice that in the `jwt` callback we pass both the access_token and the role but in the session callback we leave out the access_token and we only pass the role since this will expose jwt on the frontend through the session object. This would be the recommendable approach, but you can pass it to the session if needed.
```

Comment:
```suggestion
Notice that in the `jwt` callback we pass both the `access_token` and the `role` but in the session callback we leave out the `access_token` and we only pass the role since this would expose the token on the frontend through the session object. Exposing the `access_token` is not recommended, but you can pass it to the session if needed.
```